django_lightweight_queue/backends/reliable_redis.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import absolute_import # For 'redis'
+
+import redis
+
+from ..job import Job
+from .. import app_settings
+
+class ReliableRedisBackend(object):
+    """
+    This backend manages a per-queue-per-worker 'processing' queue. E.g. if we
+    had a queue called 'django_lightweight_queue:things', and two workers, we
+    would have:
+      'django_lightweight_queue:things:processing:1'
+      'django_lightweight_queue:things:processing:2'
+
+    We enqueue tasks to the main queue via LPUSH, and workers grab jobs by
+    atomically popping jobs from the tail of the main queue into their
+    processing queue (via BRPOPLPUSH).
+
+    On startup we remove all jobs from any processing queues, and move to the
+    tail of the main queue, i.e. so they're processed next -- see `startup`
+    below.  This is to stop losing jobs if the number of workers is lowered
+    (e.g. if we had 2 workers, both are processing a job, we kill the queues
+    and lower the number of workers to 1, without doing this tidy up we would
+    never process the job stuck in worker 2s processing queue.)
```

Comment:
@danpalmer 6b0cfc368c2ac568a06baa6db08bdf88674639d8
@prophile Great idea, since I'm concerned about the large change in moving parts in one go, per-queue backends is a great idea a3694575ffe156d4f3bde2d2b97f6d6816a8ac27 - so we can move to it gradually.